openwisp_radius/api/views.py

```diff
@@ -159,6 +160,23 @@ def get(self, request, *args, **kwargs):
 download_rad_batch_pdf = DownloadRadiusBatchPdfView.as_view()
 
 
+class UserLanguageMixin(object):
+    def save_language(self, user):
+        languages = self._parse_language_header(self.request)
+        settings_languages = list(zip(*settings.LANGUAGES))[0]
+        for language in languages:
+            if language == user.language:
+                break
+            elif language in settings_languages:
+                user.language = language
+                user.save()
```

Comment:
we should always ensure:

- the save query is executed only if the language changed
- we call `user.save(update_fields=['language'])` to ensure the query only writes this specific field

Please keep in mind this operation should be as efficient as possible because will be executed often.